### PR TITLE
LAssailant tweaks.

### PR DESCRIPTION
**Pull request by MistakeNot4892**
Renames LAssailant to a more useful varname and adds it to disposals stuffing.
Only thing that uses this var is slime friendship so shouldn't have many ramifications.

### CODE REVIEW

**Review by comma — DISMISSED** [review]

**Comment by afterthought2** [comment]
Maybe make it a weakref since you're touching it, or otherwise handle destruction. This var is notorious for making mobs fail to GC, for obvious reasons.

**Comment by MistakeNot4892** [comment]
Uses a weakref now.

**Comment by MistakeNot4892** [comment]
That was not the comment button

**Review by comma — APPROVED** [review]

File: code/modules/mob/living/carbon/xenobiological/powers.dm [hunk 1]
```diff
@@ -101,12 +101,10 @@
...
-			if(Victim && !rabid && !attacked && Victim.LAssailant && Victim.LAssailant != Victim)
-				if(!(Victim.LAssailant in Friends))
-					Friends[Victim.LAssailant] = 1
-				else
-					++Friends[Victim.LAssailant]
-
+			if(Victim && !rabid && !attacked && Victim.last_handled_by_mob)
+				var/mob/friend = Victim.last_handled_by_mob.resolve()
+				if(istype(friend) && !QDELETED(friend) && friend != Victim)
+					Friends[friend] = Friends[friend] + 1
```
Comment (afterthought2) [inline code/modules/mob/living/carbon/xenobiological/powers.dm:107]: For a future in which you continue working on slimes: this list should also use weakrefs.

**Pull request merged**